src/tools/crawler/README.md

```diff
@@ -12,21 +12,24 @@ $ cargo run --release --features crawler --bin crawler -- --help
 OPTIONS:
     -c, --crawl-interval <CRAWL_INTERVAL>
             The main crawling loop interval in seconds [default: 5]
-
+            
+    -d, --dns-seed <DNS_SEED_ADDRS>...
+            DNS seeder addresses to use (can be used multiple times)
+            
     -h, --help
             Print help information
 
     -r, --rpc-addr <RPC_ADDR>
             If present, start an RPC server at the specified address
 
     -s, --seed-addrs <SEED_ADDRS>...
-            The initial addresses to connect to
+            The initial addresses to connect to (can be used multiple times)
```

Comment:
To make the cmd UI simpler (with fewer options) - an alternative solution could have been to allow this argument to receive both (but then we would have used the` Vec<String>` and parsed/checked the IP locally):
 - human-friendly domain names (which we would resolve internally with a DNS service)
 - raw alphanumeric IP address

Reply:
True, that could be a fine feature to recognize and parse different types of addresses. 
There are, however many use cases to handle:
1. user inputs only IP address (need to add default port);
2. user inputs IP address with port (that's current `seed-addrs` case) but need to parse it (now it's done by clap);
3. user inputs hostname - need to perform DNS lookup and append port to each returned IP;
4. user inputs hostname with port - need to perform DNS lookup and use each returned IP with given port.

Basically, DNS seeder return many addresses during lookup and each address points de facto to different host. Plain host given by name should return only one address but it's possible that it returns more than one address (which could point to the same host) - that wouldn't be desired behavior as we'd like get all A records only for DNS seeder not the plain hosts.

That's something worth considering but I'd do it in separate PR as it may involve a bit more coding.

Reply:
Added to improvement bucket 😄